Simplified/NYPLBarcodeScanningViewController.m

```diff
@@ -0,0 +1,264 @@
+	CGAffineTransform transform = CGAffineTransformMakeRotation((CGFloat) (captureRotation / 180 * M_PI));
+	[self.capture setTransform:transform];
+	[self.capture setRotation:scanRectRotation];
+	self.capture.layer.frame = self.view.frame;
+}
+
+- (void)applyRectOfInterest:(UIInterfaceOrientation)orientation {
+	CGFloat scaleVideo, scaleVideoX, scaleVideoY;
+	CGFloat videoSizeX, videoSizeY;
+	CGRect transformedVideoRect = self.view.frame;
+	if([self.capture.sessionPreset isEqualToString:AVCaptureSessionPreset1920x1080]) {
+		videoSizeX = 1080;
+		videoSizeY = 1920;
+	} else {
+		videoSizeX = 720;
+		videoSizeY = 1280;
+	}
+	if(UIInterfaceOrientationIsPortrait(orientation)) {
+		scaleVideoX = self.view.frame.size.width / videoSizeX;
+		scaleVideoY = self.view.frame.size.height / videoSizeY;
+		scaleVideo = MAX(scaleVideoX, scaleVideoY);
+		if(scaleVideoX > scaleVideoY) {
+			transformedVideoRect.origin.y += (scaleVideo * videoSizeY - self.view.frame.size.height) / 2;
+		} else {
+			transformedVideoRect.origin.x += (scaleVideo * videoSizeX - self.view.frame.size.width) / 2;
+		}
+	} else {
+		scaleVideoX = self.view.frame.size.width / videoSizeY;
+		scaleVideoY = self.view.frame.size.height / videoSizeX;
+		scaleVideo = MAX(scaleVideoX, scaleVideoY);
+		if(scaleVideoX > scaleVideoY) {
+			transformedVideoRect.origin.y += (scaleVideo * videoSizeX - self.view.frame.size.height) / 2;
+		} else {
+			transformedVideoRect.origin.x += (scaleVideo * videoSizeY - self.view.frame.size.width) / 2;
+		}
+	}
+	_captureSizeTransform = CGAffineTransformMakeScale(1/scaleVideo, 1/scaleVideo);
+	self.capture.scanRect = CGRectApplyAffineTransform(transformedVideoRect, _captureSizeTransform);
+}
+
+#pragma mark - Private Methods
+
+- (NSString *)barcodeFormatToString:(ZXBarcodeFormat)format {
+  switch (format) {
+    case kBarcodeFormatAztec:
+      return @"Aztec";
+
+    case kBarcodeFormatCodabar:
+      return @"CODABAR";
+
+    case kBarcodeFormatCode39:
+      return @"Code 39";
+
+    case kBarcodeFormatCode93:
+      return @"Code 93";
+
+    case kBarcodeFormatCode128:
+      return @"Code 128";
+
+    case kBarcodeFormatDataMatrix:
+      return @"Data Matrix";
+
+    case kBarcodeFormatEan8:
+      return @"EAN-8";
+
+    case kBarcodeFormatEan13:
+      return @"EAN-13";
+
+    case kBarcodeFormatITF:
+      return @"ITF";
+
+    case kBarcodeFormatPDF417:
+      return @"PDF417";
+
+    case kBarcodeFormatQRCode:
+      return @"QR Code";
+
+    case kBarcodeFormatRSS14:
+      return @"RSS 14";
+
+    case kBarcodeFormatRSSExpanded:
+      return @"RSS Expanded";
+
+    case kBarcodeFormatUPCA:
+      return @"UPCA";
+
+    case kBarcodeFormatUPCE:
+      return @"UPCE";
+
+    case kBarcodeFormatUPCEANExtension:
+      return @"UPC/EAN extension";
+
+    default:
+      return @"Unknown";
+  }
+}
+
+#pragma mark - ZXCaptureDelegate Methods
+
+- (void)captureResult:(ZXCapture *)__unused capture result:(ZXResult *)result {
```

Comment:
I'm not seeing where captureResult is being called from?

Reply:
Good question! This is a good opportunity to highlight the common "Delegate" design pattern on iOS:
- On line 19 of the header file, we, the writers of this class, specify that we want to adhere to the Protocol `<ZXCaptureDelegate>`. Protocols are equivalent to Interfaces in Java.
- Now the compiler will enforce that we implement the required methods from the protocol.

Where is captureResult being called from? Well we "don't care", because we only care about what happens when the framework wants to call it (i.e. the implementation, which we've just written).

The framework has a reference to our class on line 78: `self.capture.delegate = self;` .. `self` is equivalent to `this` in Java. So the framework has _delegated_ some work to us and our own class, the _delegate_ has been given responsibility to handle it.

Implicit in this pattern is that the one delegating work to others, in this case the framework, is responsible to know what to do, for instance if, somehow the delegate variable is null...